### ISSUE TITLE
Clean install/run issues

### ISSUE THREAD

**Issue by mjschranz** [kriasoft/react-starter-kit#150]
After installing/updating my global `gulp` and running `npm install` and then `gulp` from the root of the cloned repository I get the following errors:

```
/Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/webpack-core/lib/NormalModuleMixin.js:206
					throw e;
					      ^
Error: Cannot find module 'path-is-absolute'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:278:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/babel-loader/lib/resolve-rc.js:17:18)
    at Module._compile (module.js:460:26)
    at Module._extensions..js (module.js:478:10)
    at Object.require.extensions.(anonymous function) [as .js] (/Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/babel-core/lib/babel/api/register/node.js:171:7)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/babel-loader/index.js:7:17)
    at Module._compile (module.js:460:26)
    at Module._extensions..js (module.js:478:10)
    at Object.require.extensions.(anonymous function) [as .js] (/Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/babel-core/lib/babel/api/register/node.js:171:7)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at DependenciesBlock.loadPitch (/Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/webpack-core/lib/NormalModuleMixin.js:193:17)
    at DependenciesBlock.doBuild (/Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/webpack-core/lib/NormalModuleMixin.js:241:4)
    at DependenciesBlock.build (/Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/lib/NormalModule.js:83:14)
    at Tapable.Compilation.buildModule (/Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/lib/Compilation.js:122:9)
    at Tapable.<anonymous> (/Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/lib/Compilation.js:390:8)
    at Tapable.<anonymous> (/Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/lib/NormalModuleFactory.js:57:13)
    at Tapable.applyPluginsAsyncWaterfall (/Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/tapable/lib/Tapable.js:75:69)
    at Tapable.onDoneResolving (/Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/lib/NormalModuleFactory.js:37:10)
    at Tapable.onDoneResolving (/Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/lib/NormalModuleFactory.js:118:6)
    at Tapable.<anonymous> (/Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/lib/NormalModuleFactory.js:114:23)
    at /Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/async/lib/async.js:251:17
    at done (/Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/async/lib/async.js:132:19)
    at /Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/async/lib/async.js:32:16
    at /Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/async/lib/async.js:248:21
    at /Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/async/lib/async.js:572:34
    at /Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/async/lib/async.js:251:17
    at done (/Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/async/lib/async.js:132:19)
    at /Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/async/lib/async.js:32:16
    at /Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/async/lib/async.js:248:21
    at /Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/enhanced-resolve/lib/UnsafeCachePlugin.js:29:4
    at onResolved (/Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/enhanced-resolve/lib/Resolver.js:39:10)
    at innerCallback (/Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/enhanced-resolve/lib/Resolver.js:89:22)
    at loggingCallbackWrapper (/Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/enhanced-resolve/lib/createInnerCallback.js:21:19)
    at /Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/tapable/lib/Tapable.js:134:6
    at /Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/enhanced-resolve/lib/ModulesInDirectoriesPlugin.js:55:11
    at /Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/enhanced-resolve/lib/Resolver.js:191:15
    at /Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/enhanced-resolve/lib/ModulesInDirectoriesPlugin.js:46:14
    at loggingCallbackWrapper (/Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/enhanced-resolve/lib/createInnerCallback.js:21:19)
    at /Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/tapable/lib/Tapable.js:134:6
    at /Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/enhanced-resolve/lib/Resolver.js:191:15
    at /Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/enhanced-resolve/lib/ModuleTemplatesPlugin.js:41:12
    at loggingCallbackWrapper (/Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/enhanced-resolve/lib/createInnerCallback.js:21:19)
    at /Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/tapable/lib/Tapable.js:134:6
    at /Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/enhanced-resolve/lib/Resolver.js:123:21
    at /Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/enhanced-resolve/lib/Resolver.js:191:15
    at applyPluginsParallelBailResult.createInnerCallback.log (/Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/enhanced-resolve/lib/Resolver.js:104:30)
    at loggingCallbackWrapper (/Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/enhanced-resolve/lib/createInnerCallback.js:21:19)
    at /Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/tapable/lib/Tapable.js:134:6
    at Tapable.next (/Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/enhanced-resolve/lib/DirectoryDescriptionFilePlugin.js:61:40)
    at Tapable.<anonymous> (/Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/enhanced-resolve/lib/DirectoryDescriptionFilePlugin.js:71:6)
    at Storage.finished (/Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:38:16)
    at evalmachine.<anonymous>:334:14
    at /Users/mschranz/Sites/repos/oicr-gdc/react-app/node_modules/webpack/node_modules/enhanced-resolve/node_modules/graceful-fs/graceful-fs.js:102:5
    at FSReqWrap.oncomplete (evalmachine.<anonymous>:95:15)
```
If I install `path-is-absolute` manually I'm able to continue but the loaded dev server on `localhost:3002` doesn't find anything.

My node/npm versions are node @0.12.3, npm@2.9.1. Any help is appreciated.

Cheers

**Comment by mjschranz** [comment]
Nevermind about the error actually getting to the dev server. That was related to a VPN I'm running.

**Comment by steve0hh** [comment]
I am also running into similar issues..

After running:
```bash
    $ git clone -o react-starter-kit -b master --single-branch \
          https://github.com/kriasoft/react-starter-kit.git MyApp
    $ cd MyApp
    $ npm install -g gulp           # Install Gulp task runner globally
    $ npm install                   # Install Node.js components listed in ./package.json
```

Then I tried to run `gulp` and get the following error:

```bash
    ERROR in ./src/decorators/withStyles.js
    Module not found: Error: Cannot resolve 'file' or 'directory' ../../node_modules/react/lib/invariant in /Users/stevetan/code/MyApp/src/decorators
     @ ./src/decorators/withStyles.js 21:37-86

    ERROR in ./src/decorators/withStyles.js
    Module not found: Error: Cannot resolve 'file' or 'directory' ../../node_modules/react/lib/ExecutionEnvironment in /Users/stevetan/code/MyApp/src/decorators
     @ ./src/decorators/withStyles.js 25:48-108

    ERROR in ./src/decorators/withContext.js
    Module not found: Error: Cannot resolve 'file' or 'directory' ../../node_modules/react/lib/emptyFunction in /Users/stevetan/code/MyApp/src/decorators
     @ ./src/decorators/withContext.js 23:41-94

    ERROR in ./src/decorators/withViewport.js
    Module not found: Error: Cannot resolve 'file' or 'directory' ../../node_modules/react/lib/ExecutionEnvironment in /Users/stevetan/code/MyApp/src/decorators
     @ ./src/decorators/withViewport.js 31:48-108
         Asset     Size  Chunks             Chunk Names
        app.js     1 MB       0  [emitted]  main
    app.js.map  1.14 MB       0  [emitted]  main

    ERROR in ./src/core/Location.js
    .
    .
    .
    .
```
Any ideas on what did I do wrongly? 
Thanks.

**Comment by TravelingTechGuy** [comment]
+1 - getting same errors, using yo generator

**Comment by DaveCowing** [comment]
+1 same errors

**Comment by powersjcb** [comment]
+1

**Comment by kabala** [comment]
+1 same problem here

**Comment by constantx** [comment]
use this [exenv](https://github.com/JedWatson/exenv) module in place of react/lib/ExecutionEnvironment

**Comment by steve0hh** [comment]
@nobesnickr has a pull request to fix this issue, #185 . tried that fork and it is working..

**Comment by koistya** [comment]
Fixed in 0e6a5203e6fa556db2e4769db839701ae82de7d6

**Comment by prescottprue** [comment]
@koistya +1 for fix with [0e6a520](https://github.com/kriasoft/react-starter-kit/commit/0e6a5203e6fa556db2e4769db839701ae82de7d6)

**Comment by langpavel** [comment]
Old, closing.
